ceremonies/src/lib.rs

```diff
@@ -359,148 +414,326 @@ decl_error! {
+            meetup_index_count[meetup_index as usize] += 1;
+            if meetup_index_count[meetup_index as usize] > meetup_index_count_max{
+                return false;
+            }
+        }
+        true
+    }
+
+
+    fn get_random_nonzero_group_element(m: u64) -> u64{
         let mut random_source = RandomNumberGenerator::<T::Hashing>::new(
-            // we don't need to pass a subject here, as this is only called once in a block.
-            // However, without subject this should be initialized outside the community loop, otherwise
-            // every community gets the same sequence of permutations if there are the same amount of
-            // people per category (bootstrappers, reputables, etc.).
             T::RandomnessSource::random_seed().0
         );
 
-        for cid in cids.iter() {
-            let pcount = <ParticipantCount>::get((cid, cindex));
-            let ecount = <EndorseesCount>::get((cid, cindex));
-
-            let n_locations = <encointer_communities::Module<T>>::get_locations(cid).len();
-
-
-
-            let mut bootstrappers =
-                Vec::with_capacity(<encointer_communities::Module<T>>::bootstrappers(cid).len());
-            let mut reputables = Vec::with_capacity(pcount as usize);
-            let mut endorsees = Vec::with_capacity(ecount as usize);
-            let mut newbies = Vec::with_capacity(pcount as usize);
-
-            // TODO: upfront random permutation
-            for p in 1..=pcount {
-                let participant = <ParticipantRegistry<T>>::get((cid, cindex), &p);
-                if Self::participant_reputation((cid, cindex), &participant)
-                    == Reputation::UnverifiedReputable
-                {
-                    reputables.push(participant);
-                } else if <encointer_communities::Module<T>>::bootstrappers(cid)
-                    .contains(&participant)
-                {
-                    bootstrappers.push(participant);
-                } else if <Endorsees<T>>::contains_key((cid, cindex), &participant) {
-                    endorsees.push(participant)
-                } else {
-                    newbies.push(participant);
+        // random number in [1, m-1]
+        (random_source.pick_usize((m - 2) as usize) + 1) as u64
+    }
+
+    fn generate_assignment_function_params(num_participants: u64, num_meetups: u64) -> AssignmentParams{
+        let max_skips = 200;
+        let m = Self::find_prime_below(num_participants);
+        let mut skip_count = 0;
+        let mut s1;
+        let mut s2;
+        loop {
+            s1 = Self::get_random_nonzero_group_element(m);
+            s2 = Self::get_random_nonzero_group_element(m);
+            if Self::validate_equal_mapping(num_participants, m, num_meetups, s1, s2) {
+                break;
+            } else {
+                skip_count += 1;
+                if skip_count > max_skips {
+                    break;
                 }
```

Comment:
this is a while loop with the condition (skip_count > max_skips) isn't it?

Reply:
well, yes with the condition `skip_count <= max_skips` though. but absolutely, idk what i was doing here😂